src/ITensorAutoHOOT/zygote.jl

```diff
@@ -0,0 +1,125 @@
+using Zygote, AutoHOOT
+using Zygote: @adjoint
+using ITensors: setinds
+
+const ad = AutoHOOT.autodiff
+const go = AutoHOOT.graphops
+
+# represent a sum of tensor networks
+mutable struct NetworkSum
```

Comment:
Does this need to be mutable?

Reply:
I have some functions that update the NetworkSum.nodes, for example see line 36

Reply:
On line 36 you are only mutating the `Array` storage of the `NetworkSum` object, not the `NetworkSum` object itself. So for example it is the difference between doing:
```julia
net_sum.nodes = push!(copy(net_sum.nodes), node)
```
and
```julia
push!(net_sum.nodes, node)
```
The second one (which is what you did) does not require making the `NetworkSum` type mutable.

In general I find it is best to make things immutable unless there is a good reason, and especially if a struct already contains a mutable data structure (like in this case) there usually isn't a reason to make it mutable.

Reply:
Done